tool.js

```diff
@@ -0,0 +1,59 @@
+// Brett Rogers (Kloe)
+
+const generator = obj => {
+    let formString = '', error = '';
+    // Checking for the settings in the object
+    if (obj.settings) {
+        formString += `<form action="${obj.settings.action}" method="${obj.settings.method}">`;        
+    } else {
+        formString += "<form action=\"#\" method=\"POST\">";
+    }
+    if (obj.tags) {
+        obj.tags.forEach(e => {
+            // Switch statement to find which tag is wanted
+            switch (e.tag) {
+                case 'input':
+                    // Checks if they added a name
+                    if (e.name) {
+                        formString += `<input type="${e.type}" name="${e.name}" />`;
+                    // Checks if they added an id
+                    } else if (e.id) {
+                        formString += `<input type="${e.type}" id="${e.id}" />`;
+                    // Checks if they added an id and name
+                    } else if (e.id && e.name) {
```

Comment:
Take a second look at your if statement, I don't think it will run the block for `else if (e.id && e.name)`.

Looks like once it finds the name attribute it just continues to next case.

Reply:
You're right! I didn't notice that part actually didn't fire. Thanks for catching that. I will definitely make that correction.